internal/cloud/s3.go

```
@@ -113,32 +113,43 @@ func (a *AWSSession) UploadCompressedDirectory(localDirectoy string, s3FilePath
 	extension := strings.ToLower(filepath.Ext(s3FilePath))
 
 	var compression archiver.Writer
-	var enc *zstd.Encoder
+	var wg sync.WaitGroup
+	var closer io.Closer
 
 	switch extension {
 	case ".xz":
+
 		compression = archiver.NewTarXz()
 		err := compression.Create(file)
 		if err != nil {
 			return err
 		}
 	case ".zstd":
 		inZstd, outTar := io.Pipe()
+		closer = outTar
 
 		compression = archiver.NewTar()
 		err := compression.Create(outTar)
 		if err != nil {
 			return err
 		}
 
-		enc, err = zstd.NewWriter(file)
+		enc, err := zstd.NewWriter(file)
 		if err != nil {
 			return err
 		}
 
+		wg.Add(1)
+
 		go func() {
-			io.Copy(enc, inZstd)
+			_, err := io.Copy(enc, inZstd)
+			if err != nil {
+				log.Println(err)
```

Shouldn't we return this error after tidying up instead of just logging it?